sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2434,829 +2447,622 @@ class CollationSQLExpressionsSuite
     )
   }
 
-  test("min_by supports collation") {
-    val collation = "UNICODE"
-    val query = s"SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);"
+  // common method for subsequent tests verifying various SQL expressions with collations
+  private def testCollationSqlExpressionCommon(
+      query: String,
+      collation: String,
+      result: Seq[Row],
+      dataType: DataType): Unit = {
```

Comment:
nit, but the parameter name could reflect that we expect the type:
```suggestion
      expectedType: DataType): Unit = {
```

Reply:
Changed to expectedTypes, as it is a sequence of DataType now